cpp/Integri/XsdComparator/XsdComparator.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <string>
+#include "Reporter.h"
+#include <map>
+#import <msxml6.dll> raw_interfaces_only
+
+class XsdComparator
+{
+public:
+	XsdComparator(Reporter* reporter);
+	void Compare(const std::string file1, const std::string file2);
+	void Compare(MSXML2::IXMLDOMElement* firstRoot, MSXML2::IXMLDOMElement* secondRoot);
+private:
+	Reporter* reporter;
```

Comment:
I wonder if this could be a reference (Reporter& reporter). You can then pass the reference to the ctor above (instead of a pointer) and thus you could avoid checking for nullptr whenever you use that.